WalletWasabi.Tests/UnitTests/Crypto/GroupElementTests.cs

```diff
@@ -274,5 +274,155 @@ public void Negation()
+		public void FromBytesThrows()
+		{
+			Assert.ThrowsAny<ArgumentException>(() => GroupElement.FromBytes(Array.Empty<byte>()));
+			Assert.ThrowsAny<ArgumentException>(() => GroupElement.FromBytes(new byte[] { 0 }));
+			Assert.ThrowsAny<ArgumentException>(() => GroupElement.FromBytes(new byte[] { 1 }));
+			Assert.ThrowsAny<ArgumentException>(() => GroupElement.FromBytes(new byte[] { 0, 1 }));
+			Assert.ThrowsAny<ArgumentException>(() => GroupElement.FromBytes(new byte[] { 0, 1 }));
+
+			Assert.ThrowsAny<ArgumentException>(() => GroupElement.FromBytes(FillByteArray(length: 32, character: 0)));
+			Assert.ThrowsAny<ArgumentException>(() => GroupElement.FromBytes(FillByteArray(length: 63, character: 0)));
+			var infinity = GroupElement.FromBytes(FillByteArray(length: 64, character: 0));
+			Assert.True(infinity.IsInfinity);
+			Assert.ThrowsAny<ArgumentException>(() => GroupElement.FromBytes(FillByteArray(length: 64, character: 1)));
+			Assert.ThrowsAny<ArgumentException>(() => GroupElement.FromBytes(FillByteArray(length: 64, character: 2)));
+			Assert.ThrowsAny<ArgumentException>(() => GroupElement.FromBytes(FillByteArray(length: 65, character: 0)));
+		}
+
+		[Fact]
+		public void Serialization()
+		{
+			var ge = GroupElement.G;
+			var ge2 = GroupElement.FromBytes(ge.ToBytes());
+			Assert.Equal(ge, ge2);
+
+			ge = GroupElement.Infinity;
+			ge2 = GroupElement.FromBytes(ge.ToBytes());
+			Assert.Equal(ge, ge2);
+
+			// Make sure infinity definition isn't f*d up in the constructor as the frombytes relies on special zero coordinates for infinity.
+			// 1. Try defining infinity with non-zero coordinates should work:
+			ge = new GroupElement(new GE(EC.G.x, EC.G.y, infinity: true));
+			byte[] zeroBytes = ge.ToBytes();
+			ge2 = GroupElement.FromBytes(zeroBytes);
+			Assert.Equal(GroupElement.Infinity, ge2);
+			// 2. Try defining non-infinity with zero coordinates should not work.
+			Assert.ThrowsAny<ArgumentException>(() => new GroupElement(new GE(FE.Zero, FE.Zero, infinity: false)));
+
+			ge = new GroupElement(EC.G * new Scalar(1));
+			ge2 = GroupElement.FromBytes(ge.ToBytes());
+			Assert.Equal(ge, ge2);
+
+			ge = new GroupElement(EC.G * new Scalar(2));
+			ge2 = GroupElement.FromBytes(ge.ToBytes());
+			Assert.Equal(ge, ge2);
+
+			ge = new GroupElement(EC.G * new Scalar(3));
+			ge2 = GroupElement.FromBytes(ge.ToBytes());
+			Assert.Equal(ge, ge2);
+
+			ge = new GroupElement(EC.G * new Scalar(21));
+			ge2 = GroupElement.FromBytes(ge.ToBytes());
+			Assert.Equal(ge, ge2);
+
+			ge = new GroupElement(EC.G * EC.NC);
+			ge2 = GroupElement.FromBytes(ge.ToBytes());
+			Assert.Equal(ge, ge2);
+
+			ge = new GroupElement(EC.G * EC.N);
+			ge2 = GroupElement.FromBytes(ge.ToBytes());
+			Assert.Equal(ge, ge2);
+		}
+
+		[Fact]
+		public void MultiplyByScalar()
+		{
+			// Scalar one.
+			var g = GroupElement.G;
+			var scalar = Scalar.One;
+			var expected = new GroupElement(EC.G * scalar);
+			Assert.Equal(expected, g * scalar);
```

Comment:
@nothingmuch @seresistvanandras ?

Reply:
since it's a typed binary operation i don't think it's wrong, but usually it's common to put the scalar on the left when using additive notation for the group, it's a strong convention.